sources/api/netdog/src/net_config/devices/bonding.rs

```diff
@@ -0,0 +1,161 @@
+use super::validate_addressing;
+use super::{error, Dhcp4ConfigV1, Dhcp6ConfigV1, Result, RouteV1, StaticConfigV1, Validate};
+use crate::interface_name::InterfaceName;
+use crate::net_config::devices::generate_addressing_validation;
+use serde::de::Error;
+use serde::{Deserialize, Deserializer};
+use snafu::ensure;
+use std::net::IpAddr;
+
+#[derive(Debug, Deserialize)]
+#[serde(deny_unknown_fields)]
+#[serde(remote = "Self")]
+pub(crate) struct NetBondV1 {
+    pub(crate) primary: Option<bool>,
+    pub(crate) dhcp4: Option<Dhcp4ConfigV1>,
+    pub(crate) dhcp6: Option<Dhcp6ConfigV1>,
+    pub(crate) static4: Option<StaticConfigV1>,
+    pub(crate) static6: Option<StaticConfigV1>,
+    #[serde(rename = "route")]
+    pub(crate) routes: Option<Vec<RouteV1>>,
+    kind: String,
+    pub(crate) mode: BondMode,
+    #[serde(rename = "min-links")]
+    pub(crate) min_links: Option<u32>,
+    #[serde(rename = "monitoring")]
+    pub(crate) monitoring_config: BondMonitoringConfig,
+    pub(crate) interfaces: Vec<InterfaceName>,
+}
+
+impl<'de> Deserialize<'de> for NetBondV1 {
+    fn deserialize<D>(deserializer: D) -> std::result::Result<Self, D::Error>
+    where
+        D: Deserializer<'de>,
+    {
+        let this = Self::deserialize(deserializer)?;
+        if this.kind.to_lowercase().as_str() != "bond" {
+            return Err(D::Error::custom(format!(
+                "kind of '{}' does not match 'bond'",
+                this.kind.as_str()
+            )));
+        }
+
+        Ok(this)
+    }
+}
+
+generate_addressing_validation!(&NetBondV1);
+
+impl Validate for NetBondV1 {
+    fn validate(&self) -> Result<()> {
+        validate_addressing(self)?;
+
+        let interfaces_count: u32 = self.interfaces.len() as u32;
+        ensure!(
+            interfaces_count > 0,
+            error::InvalidNetConfigSnafu {
+                reason: "bonds must have 1 or more interfaces specified"
+            }
+        );
+        if let Some(min_links) = self.min_links {
+            ensure!(
+                min_links <= interfaces_count,
+                error::InvalidNetConfigSnafu {
+                    reason: "min-links is greater than number of interfaces configured"
+                }
+            )
+        }
+        // Validate monitoring configuration
+        match &self.monitoring_config {
+            BondMonitoringConfig::MiiMon(config) => config.validate()?,
+            BondMonitoringConfig::ArpMon(config) => config.validate()?,
+        }
+
+        Ok(())
+    }
+}
+
+// Currently only mode 1 (active-backup) is supported but eventually 0-6 could be added
+#[derive(Clone, Debug, Deserialize)]
+#[serde(rename_all = "kebab-case")]
+pub(crate) enum BondMode {
+    ActiveBackup,
+}
```

Comment:
At the moment I'm not sure if you're using this as a number elsewhere, but if you are, this is a way that you can do it.

```suggestion
#[serde(rename_all = "kebab-case")]
#[repr(u8)]
pub(crate) enum BondMode {
    ActiveBackup = 1,
}

impl BondMode {
    pub fn as_u8(&self) -> u8 {
        self as u8
    }
}
```

Reply:
We are actually using the String right now. But we could swap to number, I believe the two are interchangeable in the wicked xml but I didn't bother checking since the strings worked. This is nice though, and if we wanted to support folks choosing either, this might also work.